src/Sylius/Bundle/CoreBundle/Validator/Constraints/OneEnabledEntityValidator.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\Validator\Constraints;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/*
+ * @author Gustavo Perdomo <gperdomor@gmail.com>
+ */
+class OneEnabledEntityValidator extends ConstraintValidator
+{
+    protected $registry;
+    protected $accessor;
+
+    public function __construct(ManagerRegistry $registry)
+    {
+        $this->registry = $registry;
+        $this->accessor = PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * @param object $entity
+     * @param Constraint $constraint
+     *
+     * @throws UnexpectedTypeException
+     * @throws ConstraintDefinitionException
+     */
+    public function validate($entity, Constraint $constraint)
+    {
+        if (!$constraint instanceof OneEnabledEntity) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__ . '\OneEnabledEntity');
+        }
+
+        $enabled = $this->accessor->getValue($entity, $constraint->enabledPath);
+
+        if ($enabled === true) {
+            return;
+        }
+
+        if ($constraint->entityManager) {
```

Comment:
That `if` from lines `54-73` can be extracted to private method `getProperEntityManager(Constraint $constraint)`